ADR/0033-upstreaming-integration-service.md

```diff
@@ -0,0 +1,228 @@
+representing the new set of images under test. [integration-service] responds, inspecting all of the
+defined [IntegrationTestScenarios], it invokes the [deployment-target-operator] (from [Decoupling
+Deployment]) to provision each cluster configuration and it runs tests of the [TestSubject] against
+each. The aggregate results and stored in the `status` of the [TestSubject] for jenkins to monitor
+and report back to the platform team in CI.
+
+**Using integration-service with TestSubject automation.** A hypothetical platform team might want
+to make use of [integration-service] in conjunction with tekton pipelines that they use for
+building images. Imagine, every night they have a cron trigger which rebuilds their images using
+a static Dockerfile that installs the latest deb packages available to a base image and then
+rebuilds all of their utility images on that new base image. Their pipelines expose the pullspecs of
+all of those images as results. The platform team configures a [ToBeDetermined] resource which
+instructed the [ToBeDetermined] controller to watch all PipelineRuns in their namespace with
+a certain label. As those builds finish, the [ToBeDetermined] controller constructs new
+[TestSubjects] by combining the image list from the **control** [TestSubject] with the new pullspec
+found in the result with the name described in the [ToBeDetermined] resource. The platform team
+benefits from integration-service's [two-phase architecture]. They build reports on the history of
+[TestSubjects] to understand when some new deb package has broken their suite of images.
+
+**Using integration-service in AppStudio** From the user's point of view, AppStudio should work like
+it does today. Our tier template needs to be modified to put a [ToBeDetermined] resource in place
+that instructs [integration-service] to construct [TestSubjects] in response to our [PaC]
+PipelineRuns.
+
+## Consequences
+
+- [integration-service] now knows nothing about [Applications], [Components], [Environments], etc.
+  It will now only depend on Tekton APIs and DeploymentTarget APIs.
+- We should now be able to tell a story about how someone might want to use [integration-service] on
+  a cluster that only carries the Tekton APIs and the DeploymentTarget APIs. The purpose of
+  integration-service is to trigger *integration tests on collections of container images*. You can
+  instruct it to do this on a case by case basis by creating TestSubject resources, or you can
+  configure a [ToBeDetermined] resource to instruct it to create its own [TestSubjects].
+- [integration-service] today has only one CustomResource of its own: [IntegrationTestScenario].
+  After this change, it will have three: [IntegrationTestScenario], [TestSubject], and
+  [ToBeDetermined].
+- We lose some ability to validate references between resources owned by different AppStudio
+  controllers. Previously, if an IntegrationTestScenario was created referencing an Application
```

Comment:
If we want to be able to re-introduce the concept of "approved entities" for modifying specific references, how might we be able to do this? I imagine it would be more critical for release-like activities which are gated by specific EnterpriseContractPolicies.

Is this really an issue? Could this be seen as a feature to enable greater abilities to reference artifacts produced outside of one's own ownership (i.e. cross-application/team/product testing)?